launcher/aboutProject/aboutproject_moc.cpp

```diff
@@ -83,3 +84,133 @@ void AboutProjectView::on_pushButtonBugreport_clicked()
 	QDesktopServices::openUrl(QUrl("https://github.com/vcmi/vcmi/issues"));
 }
 
+
+void AboutProjectView::on_pushInstallAdditional_clicked()
+{
+
+	QFileInfo file = QFileInfo(QFileDialog::getOpenFileName(this, tr("Select exe file of chronicle episode"), "", tr("Application (*.exe)")));
+	QDir sourceRoot = file.absoluteDir();
+
+	if(!sourceRoot.exists())
+		return;
+
+	int chronicleId = -1;
+	std::vector<std::string> names = { "Warlords.exe", "Underworld.exe", "Elements.exe", "Dragons.exe", "WorldTree.exe", "FieryMoon.exe", "Beastmaster.exe", "Sword.exe" };
+	for (int i = 0; i < names.size(); i++)
+	{
+		if(file.fileName().toStdString() == names[i])
+			chronicleId = i + 1;
+	}
+	if(chronicleId == -1)
+	{
+		QMessageBox::critical(this, "Chronicles data not found!", "Unknown or unsupported Chronicles found.\nPlease select directory with Heroes Chronicles.");
+		return;
+	}
+
+	extractChronicles(chronicleId, sourceRoot);
+}
+
+void AboutProjectView::extractChronicles(int chronicleId, QDir sourceRoot)
```

Comment:
Can we replace this (rather complex) logic with simple import of "xBitmap.lod" / "xSpriteLog" / etc and using them as it?

Reply:
Maybe. But instead of this import logic we have to change the region logic. With this import logic it's possible to use regions for the screen (basically the 8 icons are regions).

Reply:
Custom regions for campaigns are good idea for potential modders/map makers.

So changing campaign regions logic does not sounds like a bad idea.